app/gui/src/dashboard/modals/UpsertSecretModal.tsx

```diff
@@ -1,11 +1,9 @@
 /** @file Modal for confirming delete of any type of asset. */
 import { ButtonGroup, Dialog, DialogDismiss, Form, Input } from '#/components/AriaComponents'
+import { useSyncRef } from '#/hooks/syncRefHooks'
 import { useText } from '#/providers/TextProvider'
 import type { SecretId } from '#/services/Backend'
-
-// =========================
-// === UpsertSecretModal ===
-// =========================
+import { useEffect } from 'react'
 
 /** Props for a {@link UpsertSecretModal}. */
 export interface UpsertSecretModalProps {
```

Comment:
Why can't we just set for this component a `key`in place of use?

Reply:
See: https://react.dev/learn/preserving-and-resetting-state#resetting-a-form-with-a-key

Reply:
my bad :skull: i tried `key` on the `<Form` component and when it didn't work i figured i had to reset it manually.

... totally not realizing that of course, if the component is reused, it might not invalidate internal state